pipelines/jobs/pipeline_job_template.groovy

```diff
@@ -20,6 +20,12 @@ if (binding.hasVariable('PR_BUILDER')) {
     runSigner = false
 }
 
+// Determine releasetype choiceParam based on whether nightly or weekly-release
+releaseTypes = ['Nightly', 'Nightly Without Publish', 'Release']
+if (binding.hasVariable('WEEKLY_RELEASE') {
+    releaseTypes = ['Release']
+}
+
```

Comment:
A word of potential pessimistic warning 😅, be careful when adding if statements when fiddling around the templates.

I've been having [issues with this sort of thing recently](https://github.com/AdoptOpenJDK/openjdk-build/pull/2132#discussion_r552046743) and it's hard to notice sometimes if the parameter is being generated in the way you want it to or not.
If you find that the `releaseTypes` are not being determined correctly for each job and need help, I can give you some further info on my findings but I'm hoping you won't have the same issue I've been having 😄

Reply:
thanks for the tip, currently testing, and yes that if on releaseTypes was not working, I missunderstood what binding was. Currently testing

Reply:
all tested good now